### PR TITLE
Document tree walker class strings

### PR DESCRIPTION
**Pull request by derrabus**
Another backport from #9551

### CODE REVIEW

File: lib/Doctrine/ORM/Query/SqlWalker.php [hunk 1]
```diff
@@ -172,7 +173,9 @@ class SqlWalker implements TreeWalker
     private $quoteStrategy;
 
     /**
-     * {@inheritDoc}
+     * @param Query        $query        The parsed Query.
+     * @param ParserResult $parserResult The result of the parsing process.
+     * @psalm-param array<string, QueryComponent> $queryComponents The query components (symbol table).
```
Comment (derrabus) [inline lib/Doctrine/ORM/Query/SqlWalker.php:178]: Psalm seems to ignore `@inheritDoc` on constructors. 🤷🏻‍♂️

File: lib/Doctrine/ORM/Query/Parser.php [hunk 2]
```diff
@@ -220,7 +220,7 @@ class Parser
     /**
      * The custom last tree walker, if any, that is responsible for producing the output.
      *
-     * @var class-string<TreeWalker>
+     * @var class-string<SqlWalker>|null
```
Comment (derrabus) [inline lib/Doctrine/ORM/Query/Parser.php:223]: As explained on #9551, output walkers cannot be arbitrary implementations of `TreeWalker`, we do expect an `SqlWalker` here.

File: lib/Doctrine/ORM/Query/SqlWalker.php (hunk 1 above)
Comment (greg0ire) [reply lib/Doctrine/ORM/Query/SqlWalker.php:178]: It is supposed to work though: https://github.com/vimeo/psalm/issues/4504#issuecomment-899725382

Comment (greg0ire) [reply lib/Doctrine/ORM/Query/SqlWalker.php:178]: But I think you'd need to add `psalm-consistent-constructor` on the interface as well (or in fact, only there? not sure if it's necessary to have it in this class). Can you please try that?

Comment (derrabus) [reply lib/Doctrine/ORM/Query/SqlWalker.php:178]: Doesn't really work. If I add this annotation to the interface, `@inheritdoc` on `SqlWalker`'s constructor is still ignored. If I remove the attotation from `SqlWalker`, `@inheritdoc` works again, but this also brings back the `UnsafeInstantiation` error for `new $outputWalkerClass`. 😕

Comment (greg0ire) [reply lib/Doctrine/ORM/Query/SqlWalker.php:178]: Ok, I guess defining a constructor in an interface is quite the corner case.

**Review by greg0ire — APPROVED** [review]

**Pull request merged**